app/code/Magento/Bundle/Model/ResourceModel/Indexer/Price.php

```diff
@@ -455,6 +455,42 @@ private function getBaseBundleSelectionPriceSelect(): Select
         return $select;
     }
 
+    /**
+     * Get base select for bundle selection price update
+     *
+     * @return Select
+     * @throws \Exception
+     */
+    private function getBaseBundleSelectionPriceUpdateSelect(): Select
```

Comment:
Can you provide some details on the changes from this query and the `getBaseBundleSelectionPriceSelect`? It is important to understand your thinking on this one.

Reply:
Hi, I believe this is based on my suggested change here - https://github.com/craig-bartlett/magento2/commit/8fdd190644794887c6e5f8737789b8b8817af46f?diff=unified

I tried to explain the issue and fix briefly here https://github.com/magento/magento2/issues/12584#issuecomment-632052495 and more in depth here https://github.com/magento/magento2/pull/27315#issuecomment-630866505

Essentially it adds more ON clauses to the joins to ensure it only tries to update existing rows, fixing the issue regarding duplicate keys when saving stock as seen by @rain2o in his original pull request.

Reply:
Thanks for the info @craig-bartlett

@gabrieldagama does that help answer your question? Would you like me to add some additional comments to the docblock to make it more clear moving forward?

Reply:
It does help @rain2o, thanks @craig-bartlett. I believe that is fine, we just need to understand be reasons for the changes.